src/main/java/com/nike/cerberus/service/SafeDepositBoxService.java

```diff
@@ -390,8 +392,26 @@ public void deleteSafeDepositBox(final Set<String> groups, final String id) {
         return Optional.of(ownerPermission.get().getName());
     }
 
-    private void assertIsOwner(final Set<String> groups, final SafeDepositBoxV2 box) {
-        if (!groups.contains(box.getOwner())) {
+    private void assertIsOwner(final VaultAuthPrincipal principal, final SafeDepositBoxV2 box) {
+
+        boolean doesPrincipalHaveAdminPermissionsForSdb = false;
+        if (principal.isIamPrincipal()) {
+            Optional<Role> ownerRole = roleService.getRoleByName(RoleRecord.ROLE_OWNER);
+            for (IamPrincipalPermission perm : box.getIamPrincipalPermissions()) {
+                String roleId = perm.getRoleId();
+                Optional<Role> attachedRole = roleService.getRoleById(roleId);
+                if (attachedRole.get().getId().equals(ownerRole.get().getId())) {
+                    doesPrincipalHaveAdminPermissionsForSdb = true;
```

Comment:
nitpick - Can this variable be named "isOwner"?  
Otherwise method should be named "assertHasOwner"?